### ISSUE TITLE
Can i use my own game environnement ?

### ISSUE THREAD

**Issue by florian02100** [google/dopamine#63]
Hi, 

I want to use Dopamine with my own game environnement.
I don't use asterix, pong ou atari environnement.

Dopamine allows to do stuff like that ?

Thanks

**Comment by mgbellemare** [comment]
Yes, although at the moment you will need to modify some Atari-specific parameters (convolutional network, observation shape, etc.) I believe most of that code is in place – but stay tuned for an update that will make all of this easier.

**Comment by florian02100** [comment]
Hi Marc,

Thank you for the answer!

So I think that I need to clone atari folder (https://github.com/google/dopamine/tree/master/dopamine/atari) and I change Atari-specific parameters in preprocessing, run_experiment.py and in train.py , it's correct?

Merry Christmas !

**Comment by mgbellemare** [comment]
Yes, that's right. Look around the open/closed issues here, I believe other people have generate similar code. Good luck!

**Comment by tsachiblau** [comment]
I belive that my team has just succeed doing that.
We created new GYM env by this guide:
 [https://stackoverflow.com/questions/45068568/is-it-possible-to-create-a-new-gym-environment-in-openai](url)

this is the env class we used

`import sys
import gym
from gym import error, spaces, utils
from gym.utils import seeding
import numpy as np

class carsEnv(gym.Env):

    metadata = {"render.modes": ["human", "ansi"]}

    def __init__(self, beam_size=32, graph=None, goal_reward=10.0):
        self.screen_size=210
        self.observation_space = spaces.Box(low=0, high=255, dtype=np.uint8, shape=(84,84))
        self.action_space = spaces.Discrete(len([3]))
        self.reward_range=3
        
        
    def _step(self, action):
        reward = 0
        obs = np.zeros((84,84))
        done = False
        info = {}
        return obs, reward, done, info

    def _reset(self):
        obs = np.zeros((84,84))
        return obs


    def _render(self, mode="human", close=False):
        pass`


Hope this will help

**Comment by florian02100** [comment]
Hi tsachiblau,

Thank you ! it's a good job ! 
I'll try to create my own env from your code. 

Thank you.

**Comment by tsachiblau** [comment]
first of all you can check out our git 
[https://github.com/KatyNTsachi/Hierarchical-RL](url)
right now it's basically the original dopamine with our env in it.

If you want to see all the changes that we did you can compare our code to the original dopamin code in this link:
[https://github.com/KatyNTsachi/Hierarchical-RL/commit/40e995d9ab8cdab396415ea77c9041a53e3acbb5](url)

You also need to follow this guide for creating you own env
https://stackoverflow.com/questions/45068568/is-it-possible-to-create-a-new-gym-environment-in-openai

**Comment by ryanprinster** [comment]
@tsachiblau or any others who might know,

I just want to clarify, that we can use dopamine with our own _non-openai_ environments, as long as we can connect the observations? In my case, I am trying to use the Deepmind Lab environment.

**Comment by tsachiblau** [comment]
@ryanprinster 
you can create your our env and then wrap it with this guide:
https://stackoverflow.com/questions/45068568/is-it-possible-to-create-a-new-gym-environment-in-openai

then you will have gym env(with your own game)
